src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -76,6 +77,95 @@ def format_generic_response_payload(output: bytes, response_key: str) -> str:
         raise OpenSourceLLMUserError(message=message)
 
 
+def get_model_type(deployment_model: str) -> str:
+    m = re.match(r'azureml://registries/[^/]+/models/([^/]+)/versions/', deployment_model)
+    if m is None:
+        raise ValueError(f"Unexpected model format: {deployment_model}")
+    model = m[1].lower()
+    if model.startswith(ModelFamily.LLAMA.lower()):
+        return ModelFamily.LLAMA
+    elif model.startswith(ModelFamily.FALCON.lower()):
+        return ModelFamily.FALCON
+    elif model.startswith(ModelFamily.DOLLY.lower()):
+        return ModelFamily.DOLLY
+    elif model.startswith("gpt2"):
+        return ModelFamily.GPT2
+    else:
+        raise ValueError(f"Unexpected model type: {model} derived from deployed model: {deployment_model}")
+
+
+def get_deployment_from_endpoint(endpoint_name: str, deployment_name: str = None) -> Tuple[str, str, str]:
+    from azure.identity import DefaultAzureCredential
+    credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+
+    try:
```

Comment:
We should also consider the scenario that the user doesn't have any azure bindings locally, then we may fallback to connection?

Reply:
We're trying to avoid many users from having to create a connection.
The idea here is to allow Customers not to have to setup a connection - rather than fall back we can return an error.